starlette/decorators.py

```diff
@@ -1,10 +1,9 @@
 import asyncio
 from starlette.request import Request
-from starlette.response import Response
 from starlette.types import ASGIInstance, Receive, Send, Scope
 
 
-def asgi_application(func):
+def make_asgi(func):
```

Comment:
I'm kinda thinking that we'll drop the `asgi_application`/`make_asgi` decorator altogether - I think it just muddies things up.

Let's leave this alone for now, but perhaps don't use it in the PR, and include any logic directly in the class.